File: backend/src/main/java/com/yoohoo/backend/entity/Shelter.java
package com.yoohoo.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "Shelter")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Shelter {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long shelterId;

    @Column(nullable = false, length = 30)
    private String name;

    @Column(nullable = false, length = 30)
    private String address;

    @Column(nullable = false)
    private LocalDate foundationDate;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(nullable = false, length = 30)
    private String email;

    @Column(nullable = false, length = 30)
    private String phone;

    @Column(nullable = false, length = 30)
    private String businessNumber;

    @Column(nullable = true)
    private Integer reliability;

    @OneToMany(mappedBy = "shelter", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Dog> dogs; // 강아지 목록
}

File: frontend/src/types/shelter.ts
// 보호소 상세 정보 타입
export interface ShelterDetail {
  shelterId: number;
  name: string;
  address: string;
  foundationDate: string;
  content: string;
  email: string;
  phone: string;
  businessNumber: string;
  reliability: number;
  imageUrl: string;
  dogScore: number;
  fileScore: number;
  foundationScore: number;
}

export interface Shelter {
  shelterId: number;
  name: string;
  content: string;
  dogCount: number;
  reliability: number;
  imageUrl: string;
  foundation_date: string;
}

export interface ShelterAccountInfo {
  accountNo: string;
  bankName: string;
}

// 신뢰지수 응답 타입
export interface ReliabilityResponse {
  shelterId: number;
  reliabilityScore: number;
  dogScore: number;
  fileScore: number;
  foundationScore: number;
}
